### PR TITLE
Support PR dependent branches

### PR DESCRIPTION
**Pull request by AdamBrousseau**
- Determine if dependent string is a PR or a branch
- Fetch & checkout proper ref accodingly.

[skip ci]

Issue #827

Signed-off-by: Adam Brousseau <adam.brousseau88@gmail.com>

### CODE REVIEW

**Comment by AdamBrousseau** [comment]
jenkins compile aix jdk8 depends eclipse/omr#master

**Comment by AdamBrousseau** [comment]
jenkins compile aix jdk9 depends eclipse/omr#2457

**Comment by AdamBrousseau** [comment]
jenkins compile aix jdk10 depends eclipse/omr#2448

**Comment by AdamBrousseau** [comment]
jenkins compile aix jdk8 depends eclipse/omr#master

**Comment by AdamBrousseau** [comment]
jenkins compile aix jdk9 depends eclipse/omr#2457

**Comment by AdamBrousseau** [comment]
jenkins compile aix jdk10 depends eclipse/omr#2448

**Comment by AdamBrousseau** [comment]
`depends eclipse/omr#2457`
```
00:09:06.151 Look for Dependent Changes
[Pipeline] echo
00:09:06.278 PullRequest source repo appears to be OpenJ9
[Pipeline] echo
00:09:07.642 Dependent OMR change found: #2457
...
00:10:10.527 [omr] Running shell script
00:10:11.599 + git config remote.origin.url https://github.com/eclipse/omr.git
[Pipeline] sh
00:10:16.621 [omr] Running shell script
00:10:17.798 + git fetch --progress origin +refs/pull/2457/merge:refs/remotes/origin/pr/2457/merge
00:10:17.799 remote: Counting objects: 70, done.        
00:10:17.799 remote: Compressing objects:   6% (1/16)           
remote: Compressing objects:  12% (2/16)           
remote: Compressing objects:  18% (3/16)           
remote: Compressing objects:  25% (4/16)           
remote: Compressing objects:  31% (5/16)           
remote: Compressing objects:  37% (6/16)           
remote: Compressing objects:  43% (7/16)           
remote: Compressing objects:  50% (8/16)           
remote: Compressing objects:  56% (9/16)           
remote: Compressing objects:  62% (10/16)           
remote: Compressing objects:  68% (11/16)           
remote: Compressing objects:  75% (12/16)           
remote: Compressing objects:  81% (13/16)           
remote: Compressing objects:  87% (14/16)           
remote: Compressing objects:  93% (15/16)           
remote: Compressing objects: 100% (16/16)           
remote: Compressing objects: 100% (16/16), done.        
00:10:17.799 remote: Total 70 (delta 52), reused 63 (delta 52), pack-reused 2        
00:10:17.799 From https://github.com/eclipse/omr
00:10:17.799  * [new ref]           refs/pull/2457/merge -> origin/pr/2457/merge
[Pipeline] sh
00:10:20.921 [omr] Running shell script
00:10:21.998 + git checkout refs/remotes/origin/pr/2457/merge
00:10:21.998 Note: checking out 'refs/remotes/origin/pr/2457/merge'.
00:10:21.998 
00:10:21.998 You are in 'detached HEAD' state. You can look around, make experimental
00:10:21.998 changes and commit them, and you can discard any commits you make in this
00:10:21.998 state without impacting any branches by performing another checkout.
00:10:21.998 
00:10:21.998 If you want to create a new branch to retain commits you create, you may
00:10:21.998 do so (now or later) by using -b with the checkout command again. Example:
00:10:21.998 
00:10:21.998   git checkout -b <new-branch-name>
00:10:21.998 
00:10:21.998 HEAD is now at 849110e0... Merge 4ad5541c851120f4eca45a2f97bbf58b59048e00 into a1e32cc891effe6f6606dbe872defef859f34558
```

**Comment by AdamBrousseau** [comment]
`depends eclipse/omr#2448` (merged PR)
```
00:11:41.295 Look for Dependent Changes
[Pipeline] echo
00:11:42.675 PullRequest source repo appears to be OpenJ9
[Pipeline] echo
00:11:46.028 Dependent OMR change found: #2448
...
00:12:55.911 + git config remote.origin.url https://github.com/eclipse/omr.git
[Pipeline] sh
00:13:00.427 [omr] Running shell script
00:13:01.425 + git fetch --progress origin +refs/pull/2448/merge:refs/remotes/origin/pr/2448/merge
00:13:01.426 fatal: Couldn't find remote ref refs/pull/2448/merge
...
Setting status of 5cc2eabe038ea5cd2ef16d295eb7cf5375ff51e2 to FAILURE with url https://ci.eclipse.org/openj9/job/PullRequest-Compile-JDK10-aix_ppc-64_cmprssptrs-OpenJ9/5/ and message: 'Build finished. '
Using context: Compile - JDK10 - aix_ppc-64_cmprssptrs
ERROR: Fatal: Unable to checkout PullRequest '2448' merge commit. Either the PR does not exist, has already been merged, or there is a merge conflict.
Finished: FAILURE

**Comment by AdamBrousseau** [comment]
jenkins compile aix jdk8 depends eclipse/omr#master

**Comment by AdamBrousseau** [comment]
https://ci.eclipse.org/openj9/job/PullRequest-Compile-JDK8-aix_ppc-64_cmprssptrs-OpenJ9/12
`depends eclipse/omr#master`
```
00:08:39.134 Look for Dependent Changes
[Pipeline] echo
00:08:40.950 PullRequest source repo appears to be OpenJ9
[Pipeline] echo
00:08:44.418 Dependent OMR change found: #master
...
00:09:53.491 + git config remote.origin.url https://github.com/eclipse/omr.git
[Pipeline] sh
00:09:59.711 [omr] Running shell script
00:10:00.768 + git fetch --progress origin +refs/heads/master:refs/remotes/origin/master
[Pipeline] sh
00:10:04.291 [omr] Running shell script
00:10:05.500 + git checkout refs/remotes/origin/master
00:10:05.500 Note: checking out 'refs/remotes/origin/master'.
00:10:05.500 
00:10:05.500 You are in 'detached HEAD' state. You can look around, make experimental
00:10:05.500 changes and commit them, and you can discard any commits you make in this
00:10:05.500 state without impacting any branches by performing another checkout.
00:10:05.500 
00:10:05.500 If you want to create a new branch to retain commits you create, you may
00:10:05.500 do so (now or later) by using -b with the checkout command again. Example:
00:10:05.500 
00:10:05.501   git checkout -b <new-branch-name>
00:10:05.501 
00:10:05.501 HEAD is now at a1e32cc8... Merge pull request #2330 from dchopra001/bitPermuteZEvaluator
```

**Comment by AdamBrousseau** [comment]
ready for review @vsebe

File: buildenv/jenkins/common/build [hunk 1]
```diff
@@ -186,8 +186,22 @@ def checkout_pullrequest() {
 }
 
 def checkout_pullrequest(ID) {
-    sh "git fetch --tags --progress origin +refs/pull/*:refs/remotes/origin/pr/*"
-    sh "git checkout refs/remotes/origin/pr/${ID}/merge"
+    // Determine if we are checking out a PR or a branch
+    if(ID.isNumber()) {
```
Comment (vsebe) [inline buildenv/jenkins/common/build]: Space missing after if (...)

File: buildenv/jenkins/common/build [hunk 2]
```diff
@@ -186,8 +186,22 @@ def checkout_pullrequest() {
...
+        try {
```
Comment (vsebe) [inline buildenv/jenkins/common/build]: Reformat, use one try-catch statement.

**Comment by AdamBrousseau** [comment]
jenkins compile aix jdk8 depends eclipse/omr#master

**Comment by AdamBrousseau** [comment]
jenkins compile aix jdk9 depends eclipse/omr#2457

**Comment by AdamBrousseau** [comment]
jenkins compile aix jdk10 depends eclipse/omr#2448

**Comment by AdamBrousseau** [comment]
jenkins compile aix jdk9 depends eclipse/omr#2457

**Comment by AdamBrousseau** [comment]
jenkins compile aix jdk8 depends eclipse/omr#master

**Comment by AdamBrousseau** [comment]
@DanHeidinga or @pshipton for final review

**Pull request merged**